app/models/policy/auction.rb

```diff
@@ -0,0 +1,263 @@
+class Policy::Auction
+  BID_LIMIT = 3500
+  BID_INCREMENT = 1
+
+  include ActionView::Helpers::NumberHelper
+
+  attr_reader :auction, :user
+
+  # I don't want to make a Factory class yet
+  def self.factory(auction, user)
+    case auction.type
+    when Policy::MultiBidAuction::TYPE_CODE
+      Policy::MultiBidAuction.new(auction, user)
+    when Policy::SingleBidAuction::TYPE_CODE
+      Policy::SingleBidAuction.new(auction, user)
+    else
+      fail "Unsupported auction type: #{auction.type}"
+    end
+  end
```

Comment:
This is ugly, but I didn't want to create a separate Factory class for returning the appropriate `Policy::Auction` subtype. Should I instead not be using subclassing and mix-in base methods?